File: BackEnd/models/articleModel.js
const Joi = require('joi');

const articleSchema = Joi.object({
  title: Joi.string().required(),
  header: Joi.string().required(),
  date: Joi.date().default(Date.now),
  author: Joi.string().required(),
  body: Joi.string().required(),
  photo: Joi.string().uri().allow(''),
});

const validateArticle = (data) => {
  return articleSchema.validate(data, { abortEarly: false });
};

module.exports = { validateArticle };

File: BackEnd/models/articleModel.go
package models

import (
	"time"

	"github.com/go-playground/validator/v10"
)

// Article represents the article model
type Article struct {
	ID     int64     `json:"id"`
	Title  string    `json:"title" validate:"required"`
	Header string    `json:"header" validate:"required"`
	Date   time.Time `json:"date"`
	Author string    `json:"author" validate:"required"`
	Body   string    `json:"body" validate:"required"`
	Photo  string    `json:"photo" validate:"omitempty,url"`
}

// ValidateArticle validates the article data
func ValidateArticle(article *Article) error {
	validate := validator.New()

	// Set default date if not provided
	if article.Date.IsZero() {
		article.Date = time.Now()
	}

	return validate.Struct(article)
}
